### PR TITLE
Elena Kovalenko: weeks 1, 2

### PR DESCRIPTION
**Pull request by synthorchid**

### CODE REVIEW

File: src/new_hope/week1.clj [hunk 1]
```diff
@@ -50,3 +50,196 @@
...
+    (= (seq x) (reverse x))))
+
+(false? (palindrome? '(1 2 3 4 5)))
+(true? (palindrome? "racecar"))
+(true? (palindrome? [:foo :bar :foo]))
+(true? (palindrome? '(1 1 3 3 1 1)))
+(false? (palindrome? '(:a :b :c)))
+
+;duplicate each element in sequence
+(def duplicate
```
Comment (dryewo) [inline src/new_hope/week1.clj:200]: Try implementing this using vectors and `conj`

File: src/new_hope/week1.clj [hunk 2]
```diff
@@ -50,3 +50,196 @@
...
+      (reduce #(cons %2 (cons %2 %1)) '() coll))))
+
+(= (duplicate [1 2 3]) '(1 1 2 2 3 3))
+(= (duplicate [:a :a :b :b]) '(:a :a :a :a :b :b :b :b))
+(= (duplicate [[1 2] [3 4]]) '([1 2] [1 2] [3 4] [3 4]))
+
+;compress a sequence (remove duplicates)
+(def compress-seq
+  (fn [coll]
+    (dedupe coll)))
```
Comment (dryewo) [inline src/new_hope/week1.clj]: Well, this is cheating :) Implement it using `loop`

File: src/new_hope/week1.clj [hunk 3]
```diff
@@ -50,3 +50,196 @@
...
+
+(= (apply str (compress-seq "Leeeeeerrroyyy")) "Leroy")
+(= (compress-seq [1 1 2 3 3 2 2 3]) '(1 2 3 2 3))
+(= (compress-seq [[1 2] [1 2] [3 4] [1 2]]) '([1 2] [3 4] [1 2]))
+
+;pack sequence
+(def pack-seq
+  (fn [coll]
+    (partition-by identity coll)))
```
Comment (dryewo) [inline src/new_hope/week1.clj]: This is also meant as an exercise in using `loop`

File: src/new_hope/week1.clj [hunk 4]
```diff
@@ -50,3 +50,196 @@
...
+(= (pack-seq [1 1 2 1 1 1 3 3]) '((1 1) (2) (1 1 1) (3 3)))
+(= (pack-seq [:a :a :b :b :c]) '((:a :a) (:b :b) (:c)))
+(= (pack-seq [[1 2] [1 2] [3 4]]) '(([1 2] [1 2]) ([3 4])))
+
+;find every Nth item
+(def drop-nth
+  (fn [coll n]
+    (keep-indexed
+      (fn [i elem]
+        (if (not= 0 (mod (inc i) n)) elem nil)) coll)))
```
Comment (dryewo) [inline src/new_hope/week1.clj]: Instead of `(if x y nil)` you can use just `(if x y)` or `(when x y)`

Comment (dryewo) [inline src/new_hope/week1.clj]: Arguably better, you can also use `(pos? (mod (inc i) n))` instead of `(not= 0 (mod (inc i) n))`

**Comment by dryewo** [comment]
Please put all the tests into `(facts)` so that they are actually checked when the file is loaded.

**Comment by dryewo** [comment]
Otherwise very good 👍

File: src/new_hope/week1.clj (hunk 1 above)
Comment (synthorchid) [inline src/new_hope/week1.clj:200]: thought about it pretty much but have no idea how to do it with conj %-)

File: src/new_hope/week1.clj (hunk 3 above)
Comment (synthorchid) [inline src/new_hope/week1.clj]: done!

File: src/new_hope/week1.clj (hunk 2 above)
Comment (synthorchid) [inline src/new_hope/week1.clj]: done!

File: src/new_hope/week1.clj (hunk 1 above)
Comment (dryewo) [inline src/new_hope/week1.clj:200]: Start with replacing `'()` with `[]` and `cons` with `conj`.

**Pull request closed without merging**